backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/utils/MDevTypesUtils.java

```diff
@@ -1,48 +1,81 @@
 package org.ovirt.engine.core.common.utils;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 import org.ovirt.engine.core.common.businessentities.VM;
+import org.ovirt.engine.core.common.businessentities.VmDevice;
+import org.ovirt.engine.core.common.businessentities.VmDeviceGeneralType;
 import org.ovirt.engine.core.common.utils.customprop.SimpleCustomPropertiesUtil;
+import org.ovirt.engine.core.compat.Guid;
 import org.ovirt.engine.core.compat.Version;
 
 public class MDevTypesUtils {
 
+    public static final String DEVICE_ID = "deviceId";
+    public static final String MDEV_TYPE = "mdevType";
+    public static final String NODISPLAY = "nodisplay";
+
     /**
-     * Returns parsed mdev types from the custom properties (without the nodisplay)
+     * Returns list of mdev device spec params
      *
      * @param vm virtual machine
-     * @return the list of mdev types (without the nodisplay configuration)
+     * @return the list of mdev types
      */
-    public static List<String> getMDevTypes(VM vm) {
-        List<String> parsedMDevs = parseCustomProperties(vm);
-
-        if (!isMdevDisplayOnSupported(vm.getCompatibilityVersion())) {
-            return parsedMDevs;
+    public static List<Map<String, Object>> getMdevs(VM vm, List<VmDevice> vmDevices) {
+        boolean nodisplayPermitted = isMdevDisplayOnSupported(vm.getCompatibilityVersion());
+        List<Map<String, Object>> mdevs = new ArrayList<>();
+        for (VmDevice device : vmDevices) {
```

Comment:
I think it is safer, to avoid unwanted side effects.  We update the information (nodisplay) for further processing, but I'm not sure we want to change it the same way in the origin.  Also, we needn't store device_id in the database.

Reply:
This was eventually completely reworked.